cookbooks/docker/recipes/default.rb

```diff
@@ -0,0 +1,50 @@
+# Recipe for Docker
+
+# Install prerequisite packages
+apt_update "update-apt-cache" do
+  action :update
+end
+
+apt_package "prerequisites-linux" do
+  package_name ["curl", "linux-image-extra-" << `uname -r`, "linux-image-extra-virtual"]
+end
+
+apt_package "prerequisites-apt" do
+  package_name ["apt-transport-https", "ca-certificates"]
+end
+
+
+# Add APT key
+execute "add-apt-key" do
+  command "curl -fsSL https://yum.dockerproject.org/gpg | sudo apt-key add -"
+end
+
+execute "add-apt-repository" do
+  command 'sudo add-apt-repository "deb https://apt.dockerproject.org/repo/ ubuntu-' << `lsb_release -cs` << ' main"'
```

Comment:
Try using [apt_repository](https://docs.chef.io/resource_apt_repository.html) instead. It runs `apt-get update` too, so you can remove those code blocks as well.

For string concatenation try to use a more readable format, like [string interpolation](http://ruby-for-beginners.rubymonstas.org/bonus/string_interpolation.html). Also these thing can stored as attributes.